### ISSUE TITLE
Error on gawk when configuring on modern linux systems

### ISSUE THREAD

**Issue by NucleaPeon** [NucleaPeon/qt5-sle#4]
Error message:

```
gawk: cmd. line:22: warning: regexp escape sequence `\#' is not a known regexp operator
```

The quick fix is to install `nawk` and rearrange the qtbase/configure loop options for gawk to after nawk.
I'll be pushing a fix for this momentarily. Changing gawk's escape sequence would be preferrable, but this will document the current workaround until I get around to resolving it with gawk.

**Comment by NucleaPeon** [comment]
Line 3916 of the configure script that says
```
"$unixtests/arch.test" "$XQMAKESPEC" $OPT_VERBOSE "$relpath" "$outpath" "$OUTFILE" "target" $I_FLAGS $D_FLAGS $L_FLAGS
```

is outputting another awk error when I make the above fix

**Comment by NucleaPeon** [comment]
A couple updates.

Issue spotted in qt5-sle/qtbase/config.tests/unix/arch.test
```
awk -F: '/\=\=Qt\=magic\=Qt\=\= Architecture/ { print "'$VARPREFIX'_ARCH=\"" $2 "\"" }
             /\=\=Qt\=magic\=Qt\=\= Sub-architecture/ { print "'$VARPREFIX'_CPUFEATURES=\"" $2 "\"" }' > "$RESULTFILE"
```

I'm also seeing a `grep: warning: stray \ before c` warning in the configure script, at least when I re-run configure. Perhaps that should be a separate issue.

**Comment by NucleaPeon** [comment]
```
Determining architecture... ()
OUTDIR /home/$USER/Development/qt5-sle/qtbase
Unable to determine architecture!
```
Test is failing (edit) due to how I was running the tests. Tests aren't failing, but require the last command within the test to be a 0 exit status. 

Tested gawk and nawk with removal of the \= chars (using only `awk -F: /==` instead of `awk -F: /\=\=` for example) and that seems to work.

I modified the ./configure script to pass in the $AWK variable into the tests. I'm not sure if `awk` was used for a reason (awk calls gawk in my case) and maybe some corner cases will break by fixing these warnings and errors. Later on I will be testing this against windows xp/7 (maybe vista?) and we can see if issues persist.
